### PR TITLE
Move morton3D and sortObjects out of TreeConstruction to DetailsUtils

### PR DESCRIPTION
**Pull request by aprokop**
Few questions:
1. Is `DTK_DetailsUtils.hpp` the right place for them?
2. Should they be in `DTK` or `DTK::Details` namespace?
3. Should we put generalize `sortObjects` to work with more types, and rename to `sort`?

### CODE REVIEW

**Comment by sslattery** [comment]
1) Maybe but only if it makes sense if/when search becomes its own package
2) Probably the details namespace
3) I would say make this change later when we have a need for a different type to be sorted.

File: packages/Search/src/details/DTK_DetailsUtils.hpp [hunk 1]
```diff
@@ -351,6 +352,81 @@ typename View::non_const_type clone( View &v )
...
+    // currently assumes that the code can be represented by a 32 bit
+    // integer.
+    x = min( max( x * 1024.0, 0.0 ), 1023.0 );
+    y = min( max( y * 1024.0, 0.0 ), 1023.0 );
+    z = min( max( z * 1024.0, 0.0 ), 1023.0 );
+    unsigned int xx = expandBits( (unsigned int)x );
+    unsigned int yy = expandBits( (unsigned int)y );
+    unsigned int zz = expandBits( (unsigned int)z );
+    return xx * 4 + yy * 2 + zz;
+}
```
[the remark this answers is not in the document]
Comment (aprokop) [reply packages/Search/src/details/DTK_DetailsUtils.hpp]: What would you call it? `DTK_DetailsTreeUtils.hpp`?

[the remark this answers is not in the document]
Comment (dalg24) [reply packages/Search/src/details/DTK_DetailsUtils.hpp]: `DTK_DetailsMortonOrder.hpp` or `DTK_DetailsMortonCode.hpp`

**Comment by aprokop** [comment]
@dalg24 I addressed your comments, and introduced `DTK_DetailsSortUtils.hpp` and `DTK_DetailMortonCode.hpp`. Both Morton code and sort utilities are inside `Details` namespace. I did not address the issue of changing the `sortObjects` interface at this point.

**Pull request merged**